app.rb

```diff
@@ -42,85 +34,88 @@ def create_person
     when '2'
       create_teacher
     else
-      puts 'That is not a valid input'
-      return
+      puts 'Invalid input. Try again'
     end
-
-    puts 'Person created successfully'
   end
 
   def create_student
-    print 'Age: '
-    age = gets.chomp
-
-    print 'Name: '
+    puts 'Create a new student'
+    print 'Enter student age: '
+    age = gets.chomp.to_i
+    print 'Enter name: '
     name = gets.chomp
-
-    student = Student.new(age, name)
-    @people.push(student)
-    puts 'Student Created Successfully'
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
+    case parent_permission
+    when 'n'
+      Student.new(age, 'classroom', name, parent_permission: false)
+      @persons << student
+      puts 'Student doesnt have parent permission, cant rent books'
+    when 'y'
+      student = Student.new(age, 'classroom', name, parent_permission: false)
+      @persons << student
+      puts 'Student created successfully'
```

Comment:
- [ ] Great job so far 👍 but all requests for input and input validation should've been placed in the `main.rb` file or in a different file whose responsibilities are to request or validate input. Kindly check this [lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more information on that. Right now your `app.rb` file is full of requests for input and input validation. Kindly transfer those somewhere else so as to follow SOLID and best ruby practices 👍